sympy/concrete/summations.py

```diff
@@ -518,8 +518,9 @@ def is_convergent(self):
                 except NotImplementedError:
                     pass
 
-        ### -------------- Dirichlet tests -------------- ###
+        ### ----- Dirichlet and bounded times convergent tests ----- ###
         if order.expr.is_Mul:
+            ### -------------- Dirichlet tests -------------- ###
             a_n, b_n = order.expr.args[0], order.expr.args[1]
```

Comment:
This assumes that there are only two args to Mul. This isn't necessarily the case, for example
```
>>> O(sin(n)*log(n)/n**2, (n, oo)).expr.args
(n**(-2), log(n), sin(n))
```
As a result, `Sum(sin(n)*log(n)/n**2, (n, 1, oo)).is_convergent()` still fails to decide convergence. 

Even more troubling is the possibility that the args after the second affect the convergence of the series, which would invalidate the results of both Dirichlet test and bounded times convergent tests, as currently implemented. 

Two possible solutions: 

1. Restrict to 2-argument Mul 
2. Iterate through all possible partitions of Mul.args in two nonempty subsets, generating `a_n`, `b_n` for  the Dirichlet and bounded-convergent tests to try. This should be doable with `itertools.combinations`: say, there are N args, then loop 
```
args = order.expr.args
argset = set(args)
N = len(args)
for n in range(1, N):
    for a_tuple in itertools.combinations(args, n):
        b_set = argset - set(a_tuple)
        a_n = Mul(*a_tuple)
        b_n = Mul(*b_set)
        # call _dirichlet(a_n, b_n) and _bounded(a_n, b_n)
        # no need to try both orders of a_n and b_n, as, they both will be generated by itertools 
```

Reply:
You right. This issue was also present in the old version of the Dirichlet test.

As regards the solution you proposed, I prefer the second one, because the first seems too restrictive. Do you prefer to write the code yourself, or will I do a commit to my PR?